### DISCUSSION TITLE
Welcome to cubers.io Discussions!

### DISCUSSION THREAD

**Discussion by euphwes** [euphwes/cubers.io#150] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joaofix** [comment]
@euphwes, do you plan to open new issues in a near future? I'm looking forward to collaborating on the project but the current issues are not in my (limited) expertise domain 😜

**Comment by euphwes** [comment]
@marceloFA Definitely! I recently opened and then implemented/closed #147 (add FTO event), #148 (partial scramble previews), #134 (fixing up the settings page), and #149 (adding live preview to custom color settings).

General low-hanging fruit is just refactoring code and cleaning it up, commenting, etc. Since this is just a personal project, and not actual "work", I tend to move fast to get out new features I'm excited about.

If you have other ideas, definitely feel free to open an issue yourself, or even just submit a pull request with whatever tweaks you think improve the user experience of the site.

Other more nebulous things I've been thinking about are improving performance on the user profile pages; there's a lot of data that gets queried and then rendered there and for users that have been participating a while, it's a little slow to load. I think there's probably some combination of database query improvement (I'm sure I'm not using the ORM to its fullest capacity), and improving the profile page to ajax various components after the page loads.

Other ideas... paginating the site records pages to like 20/50/100 records at a time, so those load faster. Would require front and back-end work.

Also, I have a lot of javascript sitting directly in the templates, rather than being in javascript source files being compressed and minified. That could stand to be cleaned up, moved into source files, added to static bundles instead. I haven't opened an issue for that yet because I feel like it's mostly my mess to begin, and general code organization, cleanup, refactoring, commenting/etc are not super exciting or interesting issues.

**Comment by euphwes** [comment]
@marceloFA Opened a handful of new issues. Some of them are vague and pretty broad (general "clean this code up" sorts of stuff), but other are more concrete, manageable tasks that I don't think need a super in-depth knowledge of the app or coding experience. Take a look if you want!

**Comment by Benyo77** [comment]
@euphwes 

Hello,
I was wondering if it is intentional to have bo3 format in FMC instead of mo3. I realized this since last week I got a better mean than the guy who had beaten me with his single and this week it seems like the same happens this time favouring me.

Sorry if this isn't the right place to ask this kind of question, I'm opened to get to know the rules a bit better.

Thanks for your help!

**Comment by euphwes** [comment]
Hi @Benyo77 ,

I actually did have FMC set to have a Mo3 format when it was first introduced, but since it's a somewhat niche event, the number of weekly competitors participating in FMC is pretty low. I thought it might help to encourage more participation by lowering the "barrier to entry" and changing FMC to be Bo3 instead. That way, a casual competitor isn't required to complete all 3 solves for any solves to show up.

I know it's not technically correct, but I think it makes sense to run it that way when generally fewer people participate in that event.

**Reply by Benyo77** [reply]
Hi @euphwes 

It makes perfect sense. I was also guessing this is the intention behind having bo3, I just wanted to make sure it is not a bug or and error which accidentally changed the format without anyone noticing it.

Thanks for the nice platform and competition!
